utils/wasm-gen/src/generator/syscalls/invocator.rs

```diff
@@ -167,6 +167,25 @@ impl ParamSetter {
 
 pub type SysCallInvokeInstructions = Vec<Instruction>;
 
+pub enum SysCallKind {
+    WithDestination(usize),
+    Common,
+}
+
+impl From<InvocableSysCall> for SysCallKind {
+    fn from(invocable: InvocableSysCall) -> Self {
```

Comment:
I think it'll be better to implement method on the `InvocableSysCall`, smth like:
```rust
fn has_destination_param(&self) -> Option<usize> { ... }
```

Reply:
As `SysCallKind` looks a way more general

Reply:
done in c94c76d501cd5b3e49bf47fdf36b8943c41bfc96